app/src/main/java/io/intrepid/contest/models/Contest.java

```diff
@@ -0,0 +1,127 @@
+package io.intrepid.contest.models;
+
+import android.os.Parcel;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import timber.log.Timber;
+
+public class Contest {
+    private UUID contestId;
+    private UUID creatorId;
+    private Date creationDate;
+    private Date lastUpdatedDate;
+    private Date endedDate;
+    private String title;
+    private String description;
+    private List<Category> categories;
+
+    @Override
+    public boolean equals(Object obj) {
+        return ((obj instanceof Contest) || (obj instanceof Contest.Builder)
+                && this.contestId == ((Contest) obj).contestId
+                && this.creatorId == ((Contest) obj).creatorId
+                && this.creationDate == ((Contest) obj).creationDate);
+    }
+
+    @Override
+    public String toString() {
+        return title + "/n" +
+                description + "/n" +
+                contestId + "/n" +
+                creatorId + " " + creationDate +
+                " /n size of categories was " + categories.size() +
+                "categories " + categories +
+                " /n Last updated " + lastUpdatedDate;
+
+    }
+
+    public static class Builder {
+        public static final int UUID_MIN_LIMIT = 0;
```

Comment:
Rather then implementing this entire builder at once, lets only merge in the parts of it that are currently being used. This way when we need new feature we can simply add them to this instead of having to edit code that is already there. Remember YAGNI, we dont want to add features until we are actually using them.